Support asynchronous sending of chunked responses

fantasy-peak opened the pull request:
@tao-an 这应该是半成品，我不确定如果这样改会不会对框架造成影响，另外sendResponses那个函数似乎也需要改，但是我不确定应该如何改，你能检查一下这个mr吗

an-tao commented:
In the context of pipelining, it's possible to send multiple responses simultaneously. Subsequent responses need to wait for asynchronous responses to complete before being sent. This is a challenging issue that may require modifying the underlying Trantor interface to prevent data interleaving. @Greisby , what are your thoughts on this?

fantasy-peak commented:
对于pipelining似乎没有支持chunked的必要吧，无论怎么发最后都要按照顺序排队返回

marty1885 commented:
我得想一下，我之後寫HTTP/2 server會有點難搞。我覺的我們得有一個乾淨的API

tripleslash commented:
I modified your PR a bit to support trantors new API: https://github.com/fantasy-peak/drogon/compare/master...tripleslash:drogon:master

fantasy-peak commented:
> I modified your PR a bit to support trantors new API: [fantasy-peak/drogon@master...tripleslash:drogon:master](https://github.com/fantasy-peak/drogon/compare/master...tripleslash:drogon:master)



> I modified your PR a bit to support trantors new API: [fantasy-peak/drogon@master...tripleslash:drogon:master](https://github.com/fantasy-peak/drogon/compare/master...tripleslash:drogon:master)

You can send a new mr and I will close this mr
